### PR TITLE
add DDIM, DPM-Solver (++), UniPC samplers

### PR DESCRIPTION
**Pull request by zhaoyuzhi**
- add DDIM and UniPC codes at `ldm/models/diffusion`
- update current DPM-Solver code
- add 5 testing scripts at `scripts/tests`
- update the README.md (but I cannot upload the demo images. I will find an approach later)

### CODE REVIEW

File: examples/stable_diffusion_v2/README.md [hunk 1]
```diff
@@ -57,13 +57,63 @@ img3.jpg,a red and white ball with an angry look on its face
...
+```
+
+| PLMS | DDIM | DPM-Solver | DPM-Solver++ | UniPC |
+| :----: | :----: | :----: | :----: | :----: |
+| <img src="https://github.com/zhaoyuzhi/mindone/demo/sunflower/PLMS_1.png" width="155" height="155" /> | <img src="https://github.com/zhaoyuzhi/mindone/demo/sunflower/DDIM_1.png" width="155" height="155" /> | <img src="https://github.com/zhaoyuzhi/mindone/demo/sunflower/DPM-Solver_1.png" width="155" height="155" /> | <img src="https://github.com/zhaoyuzhi/mindone/demo/sunflower/DPM-Solver++_1.png" width="155" height="155" /> | <img src="https://github.com/zhaoyuzhi/mindone/demo/sunflower/UniPC_1.png" width="155" height="155" /> |
+| <img src="https://github.com/zhaoyuzhi/mindone/demo/sunflower/PLMS_2.png" width="155" height="155" /> | <img src="https://github.com/zhaoyuzhi/mindone/demo/sunflower/DDIM_2.png" width="155" height="155" /> | <img src="https://github.com/zhaoyuzhi/mindone/demo/sunflower/DPM-Solver_2.png" width="155" height="155" /> | <img src="https://github.com/zhaoyuzhi/mindone/demo/sunflower/DPM-Solver++_2.png" width="155" height="155" /> | <img src="https://github.com/zhaoyuzhi/mindone/demo/sunflower/UniPC_2.png" width="155" height="155" /> |
```
[the remark this answers is not in the document]
Comment (zhaoyuzhi) [reply examples/stable_diffusion_v2/README.md]: This issue has been resolved. Plz check.

**Comment by SamitHuang** [comment]
Please fix the doc conflicts. And re-organize the added samplers and their evaluation results into a dependent section. For example, Add a `Supported Samplers` section under `Inference`.


#### Supported Samplers

- PLMS
- DPM Solver
- DPM Solver++
...

Visual Comparison:

....

**Comment by zhaoyuzhi** [comment]
> Please fix the doc conflicts. And re-organize the added samplers and their evaluation results into an independent section.
> 
> For example, add a `Supported Samplers` section under `Inference`:
> 
> ```
> #### Supported Samplers
> 
> - PLMS
> - DPM Solver
> - DPM Solver++
> ...
> 
> Visual Comparison:
> ...
> ```
> 
> If the content is too long, it's better to put it in a new doc like `samplers.md` and link to main readme.md.



> Please fix the doc conflicts. And re-organize the added samplers and their evaluation results into an independent section.
> 
> For example, add a `Supported Samplers` section under `Inference`:
> 
> ```
> #### Supported Samplers
> 
> - PLMS
> - DPM Solver
> - DPM Solver++
> ...
> 
> Visual Comparison:
> ...
> ```
> 
> If the content is too long, it's better to put it in a new doc like `samplers.md` and link to main readme.md.

I have added a "schedulers.md" and update the "readme.md"

**Comment by SamitHuang** [comment]
Looks better! 
Please rebase the commits on our latest master branch to resolve the conflicts. For example, create a new branch in your repo based on official mindone,  and `git checkout master`  then  `git merge --squash your_sample_branch`

**Pull request closed without merging**